Add new modules documentation

jfy133 opened the pull request:
Transferred from the nf-core/modules README

ewels commented on markdown/developers/adding_modules.md:
```diff
@@ -0,0 +1,406 @@
+
+## Adding a new module file
+
+If you decide to upload a module to `nf-core/modules` then this will ensure that it will become available to all nf-core pipelines, and to everyone within the Nextflow community! See [`modules/`](modules)
```
Link is now wrong I think..

ewels commented on markdown/developers/adding_modules.md:
```diff
@@ -0,0 +1,406 @@
+
+The features offered by Nextflow DSL2 can be used in various ways depending on the granularity with which you would like to write pipelines. Please see the listing below for the hierarchy and associated terminology we have decided to use when referring to DSL2 components:
+
+- *Module*: A `process` that can be used within different pipelines and is as atomic as possible i.e. cannot be split into another module. An example of this would be a module file containing the process definition for a single tool such as `FastQC`. At present, this repository has been created to only host atomic module files that should be added to the [`modules/`](modules/) directory along wit ...
```
Could we make these have their own level 3 headings please? Then we can link to them directly.

ewels commented on markdown/developers/adding_modules.md:
```diff
@@ -0,0 +1,406 @@
+1. Install the latest version of [`nf-core/tools`](https://github.com/nf-core/tools#installation) (`>=2.0`)
+2. Install [`Nextflow`](https://www.nextflow.io/docs/latest/getstarted.html#installation) (`>=21.04.0`)
+3. Install any of [`Docker`](https://docs.docker.com/engine/installation/), [`Singularity`](https://www.sylabs.io/guides/3.0/user-guide/) or [`Conda`](https://conda.io/miniconda.html)
+4. [Fork and clone this repo locally](#uploading-to-nf-coremodules)
```
Needs rewriting

jfy133 replied:
Nore sure what you mean here?

ewels replied:
"This repo" makes sense when in a readme of the modules repo. Not when in standalone docs in the website though.

ewels reviewed (changes requested):
Looking good! Few text tweaks needed. I started going through making suggested changes for lots of heading levels (number of `#`s) as it's much more important for the website than it is on GitHub. But then I changed my mind and thought I should see how it renders first. But they may need some attention.

jfy133 commented:
I've just updated the headers anyway because I shuffled a few sections around, and I replaced all references to repository directories with URLS (I think I got them all, but please let me know if you identify more)

I also added an additional step in the workflow to check guidelines before linting.

ewels commented on markdown/developers/adding_modules.md:
```diff
@@ -0,0 +1,413 @@
+2. Install [`Nextflow`](https://www.nextflow.io/docs/latest/getstarted.html#installation) (`>=21.04.0`)
+3. Install any of [`Docker`](https://docs.docker.com/engine/installation/), [`Singularity`](https://www.sylabs.io/guides/3.0/user-guide/) or [`Conda`](https://conda.io/miniconda.html)
+4. [Fork and clone this repo locally](#uploading-to-nf-coremodules)
+5. Set up git by adding a new remote of the nf-core git repo called `upstream`
```
```suggestion
5. Set up git on your computer by adding a new git remote of the main nf-core repo called `upstream`
```

ewels reviewed (approved):
LGTM on my phone! Good improvements ✅ Just the reference to "this repo" remaining that I can see from a quick skim read.

jfy133 commented:
@drpatelh any thoughts before I merge?

drpatelh reviewed (approved):
LGTM! These should evolve over time anyway but defo a good start! Thanks @jfy133 ! Apologies for the delay.
